rvzr/model_dynamorio/backend/include/util.hpp

```diff
@@ -33,3 +35,21 @@ void reserve_register_checked(void *drcontext, instrlist_t *ilist, instr_t *wher
 /// @param reg The register to unreserve
 /// @return void
 void unreserve_register_checked(void *drcontext, instrlist_t *ilist, instr_t *where, reg_id_t reg);
+
+/// @brief Write to the given addres ignoring any permission check.
+/// NOTE: Writing to executable memory might clash with DynamoRIO's code cache, use with caution.
+/// @param addr address to write to
+/// @param size size of the write
+/// @param val pointer to the value to write
+/// @param w_size number of bytes written is stored here (see dr_safe_write)
+/// @return true if the write was successfull
+bool force_write(byte *addr, size_t size, const uint64_t *val, size_t *w_size);
+
+/// @brief Check if an instruction is an indirect call or return to an illegal instruction.
+///        This has to be called from _within_ the dispatcher's clean_call to make sure that the
+///        runtime value of the target is known.
+/// NOTE: This operation is not cheap, use with caution.
+/// @param instr The instruction to check
+/// @param mc
+/// @param dc
+bool is_illegal_jump(instr_obs_t instr, dr_mcontext_t *mc, void *dc);
```

Comment:
It would be helpful to leave a note in the docstring mentioning that this function is unused, and why.